file.go

```diff
@@ -0,0 +1,205 @@
+package cfile
+
+// TODO:
+// * explore file locking
+// * use mmap, but that would make appending/trunc a lot harder.
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"sync"
+	"sync/atomic"
+)
+
+var (
+	_ io.Reader     = (*File)(nil)
+	_ io.ReaderFrom = (*File)(nil)
+	_ io.ReaderAt   = (*File)(nil)
+	_ io.Writer     = (*File)(nil)
+	_ io.WriterTo   = (*File)(nil)
+	_ io.WriterAt   = (*File)(nil)
+	_ io.Closer     = (*File)(nil)
+)
+
+// New opens `fp`` with `os.O_CREATE|os.O_RDWR`` and the given permissions.
+func New(fp string, perm os.FileMode) (*File, error) {
```

Comment:
Is there any way we can utilize named returns? Our codebase generally follows the format, if you can use a named return - please do so.

Reply:
Using named returns here would be kinda pointless imo i'd have to split the declarations into vars.

named returns are only useful when there's ambiguity about the returned values.